TestFoundation/TestNSStream.swift

```diff
@@ -124,6 +132,107 @@ class TestNSStream : XCTestCase {
         XCTAssertEqual(Stream.Status.error, fileStream.streamStatus)
     }
     
+    /* This test is the success case for Streams that are in the blessed list with the
+     * dataGetBuffer as part of the call back e.g
+     * static const struct _CFStreamCallBacksV1 readDataCallBacks = {1, readDataCreate,readDataFinalize, readDataCopyDescription, readDataOpen, NULL, dataRead, dataGetBuffer, dataCanRead, NULL, NULL, NULL, NULL, NULL, NULL, readDataSchedule, NULL};
+     */
+    func test_InputStreamGetBufferSuccessFromBlessedList(){
+        let message: NSString = "Hello, playground"
+        let messageData: Data = message.data(using: String.Encoding.utf8.rawValue)!
+        let dataStream = InputStream(data: messageData)
+        XCTAssertEqual(Stream.Status.notOpen, dataStream.streamStatus)
+        dataStream.open()
+        XCTAssertEqual(Stream.Status.open, dataStream.streamStatus)
+        let buffer = UnsafeMutablePointer<UnsafeMutablePointer<UInt8>?>.allocate(capacity: 1)
```

Comment:
I don't see anything holding up this patch. It's fine as-is.

I've never looked at this API before, but I think it would be a simplification to change the signature of getBuffer to return an optional raw pointer and length:
func getBuffer() -> (UnsafeMutableRawPointer, Int)?

What's the point in returning a bool?
 
If-and-when I propose UnsafeMutableRawBufferPointer, that would be a better return type. I'll try to spend some time on that in the next couple days, but it doesn't need to hold up this PR.